x-pack/test/functional_execution_context/test_utils.ts

```diff
@@ -41,12 +50,89 @@ export async function assertLogContains({
 }): Promise<void> {
   // logs are written to disk asynchronously. I sacrificed performance to reduce flakiness.
   await retry.waitFor(description, async () => {
-    const logsStr = await Fs.readFile(logFilePath, 'utf-8');
-    const normalizedRecords = logsStr
-      .split(endOfLine)
-      .filter(Boolean)
-      .map((s) => JSON.parse(s));
-
-    return normalizedRecords.some(predicate);
+    if (!logstream$) {
+      logstream$ = getLogstream$();
+    }
+    try {
+      await firstValueFrom(logstream$.pipe(filter(predicate), timeout(5_000)));
+      return true;
+    } catch (err) {
+      return false;
+    }
   });
 }
+
+/**
+ * Creates an observable that continuously tails the log file.
+ */
+function getLogstream$(): ReplaySubject<Ecs> {
+  const stream$ = new ReplaySubject<Ecs>();
+
+  defer(async function* () {
+    const fd = await Fs.open(logFilePath, 'rs');
+    while (!stream$.isStopped) {
+      const { bytesRead, buffer } = await fd.read();
+      if (bytesRead) {
+        yield buffer.toString('utf8', 0, bytesRead);
+      }
+    }
+    await fd.close();
```

Comment:
Using loop with `fd.read` instead of the `createReadStream` because the latter closes the stream when it gets to the EOF. I want to keep tailing until we close the stream.